driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/quotactl.bpf.c

```diff
@@ -0,0 +1,253 @@
+	struct if_dqblk dqblk = {0};
+	if(scap_cmd == PPM_Q_GETQUOTA || scap_cmd == PPM_Q_SETQUOTA)
+	{
+		bpf_probe_read_user((void *)&dqblk, sizeof(dqblk), (void *)addr_pointer);
+	}
+
+	/* Please note that `dqblk` struct could be filled with values different from `0`,
+	 * even if these values are not valid, so we need to explicitly send `0`.
+	 */
+	if(dqblk.dqb_valid & QIF_BLIMITS)
+	{
+		/* Parameter 4: dqb_bhardlimit (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, dqblk.dqb_bhardlimit);
+
+		/* Parameter 5: dqb_bsoftlimit (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, dqblk.dqb_bsoftlimit);
+	}
+	else
+	{
+		/* Parameter 4: dqb_bhardlimit (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, 0);
+
+		/* Parameter 5: dqb_bsoftlimit (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, 0);
+	}
+
+	if(dqblk.dqb_valid & QIF_SPACE)
+	{
+		/* Parameter 6: dqb_curspace (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, dqblk.dqb_curspace);
+	}
+	else
+	{
+		/* Parameter 6: dqb_curspace (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, 0);
+	}
+
+	if(dqblk.dqb_valid & QIF_ILIMITS)
+	{
+		/* Parameter 7: dqb_ihardlimit (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, dqblk.dqb_ihardlimit);
+
+		/* Parameter 8: dqb_isoftlimit (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, dqblk.dqb_isoftlimit);
+	}
+	else
+	{
+		/* Parameter 7: dqb_ihardlimit (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, 0);
+
+		/* Parameter 8: dqb_isoftlimit (type: PT_UINT64) */
+		auxmap__store_u64_param(auxmap, 0);
+	}
+
+	if(dqblk.dqb_valid & QIF_BTIME)
+	{
+		/* Parameter 9: dqb_btime (type: PT_RELTIME) */
+		auxmap__store_u64_param(auxmap, dqblk.dqb_btime);
+	}
+	else
+	{
+		/* Parameter 9: dqb_btime (type: PT_RELTIME) */
+		auxmap__store_u64_param(auxmap, 0);
+	}
+
+	if(dqblk.dqb_valid & QIF_ITIME)
+	{
+		/* Parameter 10: dqb_itime (type: PT_RELTIME) */
+		auxmap__store_u64_param(auxmap, dqblk.dqb_itime);
+	}
+	else
+	{
+		/* Parameter 10: dqb_itime (type: PT_RELTIME) */
+		auxmap__store_u64_param(auxmap, 0);
+	}
+
+	/* We extract the `struct if_dqinfo` if possible. */
+	struct if_dqinfo dqinfo = {0};
+	if(scap_cmd == PPM_Q_GETINFO || scap_cmd == PPM_Q_SETINFO)
+	{
+		bpf_probe_read_user((void *)&dqinfo, sizeof(dqinfo), (void *)addr_pointer);
+	}
+
+	if(dqinfo.dqi_valid & IIF_BGRACE)
+	{
+		/* Parameter 11: dqi_bgrace (type: PT_RELTIME) */
+		auxmap__store_u64_param(auxmap, dqinfo.dqi_bgrace);
+	}
+	else
+	{
+		/* Parameter 11: dqi_bgrace (type: PT_RELTIME) */
+		auxmap__store_u64_param(auxmap, 0);
+	}
+
+	if(dqinfo.dqi_valid & IIF_IGRACE)
+	{
+		/* Parameter 12: dqi_igrace (type: PT_RELTIME) */
+		auxmap__store_u64_param(auxmap, dqinfo.dqi_igrace);
+	}
+	else
+	{
+		/* Parameter 12: dqi_igrace (type: PT_RELTIME) */
+		auxmap__store_u64_param(auxmap, 0);
+	}
+
+	if(dqinfo.dqi_valid & IIF_FLAGS)
+	{
+		/* Parameter 13: dqi_flags (type: PT_FLAGS8) */
+		auxmap__store_u8_param(auxmap, dqinfo.dqi_flags);
```

Comment:
Probably yet another missing `scap` mapper for `dqi_flags` for #515 .

Reply:
Nice to see you again !! :rocket: Added! thank you

Reply:
:-) Looked thru your commits... and they are all nice. Catching up with the others PRs over the next days.